crypto/dsa/dsa_check.c

```diff
@@ -19,15 +19,39 @@
 #include "dsa_local.h"
 #include "crypto/dsa.h"
 
-int dsa_check_params(const DSA *dsa, int *ret)
+#ifdef FIPS_MODULE
+int dsa_check_params(const DSA *dsa, int checktype, int *codes)
 {
     /*
      * (2b) FFC domain params conform to FIPS-186-4 explicit domain param
      * validity tests.
      */
     return ossl_ffc_params_FIPS186_4_validate(dsa->libctx, &dsa->params,
-                                              FFC_PARAM_TYPE_DSA, ret, NULL);
+                                              FFC_PARAM_TYPE_DSA, codes, NULL);
 }
+#else
+int dsa_check_params(const DSA *dsa, int checktype, int *codes)
+{
+    FFC_PARAMS params = {0};
+    int ret;
+
+    if (dsa->params.seed != NULL)
```

Comment:
@t8m I am not sure this is the right approach here.
I imagined the code to look something like

````
   if (checktype == quick)
      simple_validate();
   else
      full_validate();
````

The app can then only do the quick validate for dsa.

I think with the simple_check we then just need to decide if we want to do the thing that you are deciding with the check_type parameter currently.

Reply:
The app should do the full check, if at all possible. And I do not see a reason why the full check could not mean something else with the default provider and the fips provider.

Yes, I can organize the things a little differently - i.e. make the simple_validate to be a quick check and to make the full check to be a separate function. I do not have a problem with that.